client/landing/stepper/declarative-flow/internals/steps-repository/plans/plans-wrapper.tsx

```diff
@@ -72,11 +76,19 @@ const PlansWrapper: React.FC< Props > = ( props ) => {
 			 ).getHidePlansFeatureComparison(),
 		};
 	}, [] );
-	const { flowName } = props;
+	const { flowName, selectedSiteId, setSelectedSiteId } = props;
 
 	const { setPlanCartItem, setDomain, setDomainCartItem } = useDispatch( ONBOARD_STORE );
 
 	const site = useSite();
+	const siteId = site?.ID;
```

Comment:
I see that selectedSiteId _is already being passed_ through to the plans features main, the component responsible for rendering the plans grid in Calypso

https://github.com/Automattic/wp-calypso/blob/2f40b62fd783637f4655d225efc19e5eb8aeba30/client/landing/stepper/declarative-flow/internals/steps-repository/plans/plans-wrapper.tsx#L141-L145

I think what you have here could work fine as a patch, but it seems like PlansFeaturesMain isn't respecting the selectedSiteId, even when it's a meaningful value. 🤔

Reply:
Here's a rough PR of what I'm describing https://github.com/Automattic/wp-calypso/pull/83785. We utilize the `siteId` that's already being passed as a prop to `PlansFeaturesMain` as a fallback when evaluating selectedSiteId in the plans grid

Reply:
Good thinking @jeyip . I wonder why we need to pass the site ID from here if `plans-features-main` can query it from the Calypso monostore like you said? Is there a chance the two might differ - the onboard store has something else stored? 🤔

Reply:
I am not sure that we can safely assume that the Calypso data store is current when we're running in Stepper, even if it so happens that they're probably aligned for this specific situation.

Reply:
Just to confirm / and apologies for being naive / we _do_ need the site ID context for the plans grid in Stepper right? and anything from `plans-features-main` and beyond uses Calypso monostore state around it.

Reply:
> we do need the site ID context for the plans grid in Stepper right?

From what I can tell, this onboarding view is a special case. Typically we _don't_ need a siteID since a site is still in the process of being created in Stepper, but in this scenario, the onboarding flow is displayed in a _post_ signup context.

Ex. A user creates their site. They are redirected to the Launchpad PCYsg-LPB-p2 ( which is rendered in Stepper ). The Launchpad attempts to upsell the user on a plan ( if they initially created a free site ) and redirects them to a Stepper flow to do so ( domain upsell ).

The logic below then creates the expectation in `PlansFeaturesMain` that there will be a meaningful siteID provided to the component

https://github.com/Automattic/wp-calypso/blob/a8f194a565ae41412dee23fb5b1eeb66e7eb4154/client/landing/stepper/declarative-flow/internals/steps-repository/plans/plans-wrapper.tsx#L86